blueocean-pipeline-editor/src/main/js/components/editor/EditorPipelineGraph.jsx

```diff
@@ -362,42 +369,50 @@ needsLayout = true;
     }
 
     renderBigLabel(details:LabelInfo) {
-
         const { nodeSpacingH, labelOffsetV } = this.state.layout;
-
+        const stage = details.parentStage && details.parentStage.children || details.stage;
+        const isTopLevelParallel = stage.children.length;
         const labelWidth = nodeSpacingH;
         const labelOffsetH = Math.floor(labelWidth * -0.5);
 
-        // These are about layout more than appearance, so they should probably remain inline
-        const bigLabelStyle = {
-            position: "absolute",
-            width: labelWidth,
-            textAlign: "center",
-            marginLeft: labelOffsetH,
-            marginBottom: labelOffsetV
-        };
-
         const x = details.x;
         const bottom = this.state.measuredHeight - details.y;
 
-        const style = Object.assign({}, bigLabelStyle, {
+        const style = {
+            width: labelWidth,
+            marginLeft: labelOffsetH,
+            marginBottom: labelOffsetV - 4,
             bottom: bottom + "px",
             left: x + "px"
-        });
+        };
 
-        const stage = details.stage;
         const key = (stage ? stage.id : details.text) + "-big"; // TODO: Replace with a key on LabelInfo
+        const inner = [];
 
         const classNames = ["pipeline-big-label"];
-        if (this.nodeIsSelected(details.node)
-            || (stage && this.stageChildIsSelected(stage))) {
+        if (this.selectedStage === stage) {
             classNames.push("selected");
         }
-        if (nodeHasErrors(details.node)) {
+        if (isTopLevelParallel) {
+            classNames.push('top-level-parallel');
+            // add a top-level parallel config icon
+            inner.push(<Icon icon="NavigationMoreHoriz" size={24} />);
+            inner.push(<Icon icon="NavigationMoreHoriz" size={24} />);
+        }
+        // add an alert if the stage has errors
+        if (stageHasDirectErrors(stage)) {
             classNames.push("errors");
+            if (isTopLevelParallel) {
+                inner.push(<AlertIcon />);
+            }
         }
-
-        return <div className={classNames.join(" ")} style={style} key={key}>{details.text}</div>;
+        return (
+            <div className={classNames.join(" ")} style={style} key={key}
+                 onClick={e => this.nodeClicked({ isPlaceholder: false, stage }, e)}>
+                {details.text || '\u00a0'}
```

Comment:
It might be good to assign this value to a constant with helpful name like `NON_BREAKING_SPACE`. I had to google the char sequence to see what it was.